rclrs_examples/src/minimal_publisher.rs

```diff
@@ -17,13 +13,11 @@ fn main() -> Result<(), Error> {
 
     let mut publish_count: u32 = 1;
 
-    while context.ok()? {
+    loop {
```

Comment:
I was going to say something here about how this change doesn't allow a graceful wind down for the node, but I just tested the `master` branch and found that it doesn't gracefully wind down either. I wonder if `rclcpp` does something special that `rcl` itself doesn't do to allow its `Context::ok()` to know whether `SIGINT` has been triggered.

Reply:
Yes, `rclcpp` registers a signal handler (see https://github.com/ros2/rclcpp/blob/master/rclcpp/src/rclcpp/signal_handler.hpp) to make sure the context is shutdown correctly (and also for other stuff, like logging). I'd keep the `ok` function and improve it to work correctly with `SIGINT`

Reply:
In that case I wonder whether observing `SIGINT` should really be done with `Context::ok` or if we should consider a different way of observing `SIGINT`.

My current impression is that there are only two things that can make `rclcpp::Context::ok` false:
1. `SIGINT` is received
2. `rclcpp::shutdown(context)` is called

If that's the case then I might float the idea of having a singleton independent of `rclrs::Context` whose entire purpose is to reflect whether `SIGINT` has occurred.

But if there are other potential reasons that `Context::ok` could turn out to be false (e.g. issues from the RMW layer?) then I think bringing back `Context::ok` and having it reflect `SIGINT` status makes sense.

Reply:
Would it be a good idea to do signal handling in the `spin()` method?

Reply:
A more general thing to do for `spin` would be to have an argument like

```
keep_spinning: Box<dyn FnMut() -> bool>
```

and have that callback triggered on each cycle to decide whether to keep spinning. Then the user can customize the decision however they'd like.

But if that's considered too complicated for the basic `spin` function, then yes I think it should at least be listening for `SIGINT`.

Reply:
The `spin` function will removed eventually to support executors. I'd put the signal handling in the context (like `rclcpp` and `rclpy` do), so that we can register callbacks and do logging from there for example.

Reply:
So would you like me to add signal handling in this PR?

If not, I can add back the `Context::ok()` function and make it always return `true`, since `rcl_context_is_valid()` is redundant (we know that it hasn't been shutdown since the value hasn't been dropped). That doesn't feel too useful though.

Reply:
Don't worry about signal handling now, just add `ok` back and we can improve that later. Thanks!

Reply:
Added it back.

Reply:
Can you add it back as it was not just a skeleton? Thanks.

Reply:
That's what I was talking about above. The call to `rcl_context_is_valid()` is entirely redundant. Only `rcl_shutdown()` can cause that call to return `false`, but we know shutdown hasn't been called (since shutdown only happens when the context is dropped).

Reply:
Still, that's not a reason for removing it, the signal handler will most likely registered outside the `ok` funcion, which will end up being like it is currently on `main`. Please add the code back.

Reply:
Yes, hence why the call to `rcl_context_is_valid` still needs to be in `ok`

Reply:
Done.